src/feedback/tests/test_views_public.py

```diff
@@ -228,3 +230,104 @@ def test_superuser(self):
+                                                sichtbarkeit='ADM')
+
+        self.veranstaltung = Veranstaltung.objects.create(
+            typ='v', name='GDI',
+            semester=self.semester,
+            grundstudium=False,
+            evaluieren=True
+        )
+
+        self.deleted_veranstaltung = Veranstaltung.objects.create(
+            typ='v', name='GDI2',
+            semester=self.semester,
+            grundstudium=False,
+            evaluieren=True
+        )
+
+        self.last_state_veranstaltung = Veranstaltung.objects.create(
+            typ='v', name='GDI3',
+            semester=self.semester,
+            grundstudium=False,
+            evaluieren=True,
+            status=Veranstaltung.STATUS_BOEGEN_GESCANNT
+        )
+        self.deleted_barcode = self.deleted_veranstaltung.get_barcode_number()
+        self.deleted_veranstaltung.delete()
+
+    def assertJsonSuccess(self, response, success):
+        response_json = json.loads(response.content)
+        self.assertEqual(response_json['success'], success)
+
+    def test_invalid_token(self):
+        response = self.client.post(self.path, {"barcode": 22})
+        self.assertJsonSuccess(response, False)
+
+        response = self.client.post(self.path, {"scanner_token": ""})
+        self.assertJsonSuccess(response, False)
+
+    def test_invalid_barcode(self):
+        response = self.client.post(self.path, {"scanner_token": self.barcode_scanner.token})
+        self.assertJsonSuccess(response, False)
+
+        response = self.client.post(self.path, {"barcode": 22, "scanner_token": self.barcode_scanner.token})
+        self.assertJsonSuccess(response, False)
+
+        response = self.client.post(self.path, {
+            "barcode": self.deleted_barcode,
+            "scanner_token": self.barcode_scanner.token
+        })
+        self.assertJsonSuccess(response, False)
+
+    def test_barcode_scanner_right(self):
+        barcode = self.veranstaltung.get_barcode_number()
+        response = self.client.post(self.path, {
+            "barcode": barcode,
+            "scanner_token": self.barcode_scanner_no_access_right.token
+        })
+        self.assertJsonSuccess(response, False)
+        
+        barcode = self.last_state_veranstaltung.get_barcode_number()
+        response = self.client.post(self.path, {
+            "barcode": barcode,
+            "scanner_token": self.barcode_scanner.token
+        })
+        self.assertJsonSuccess(response, False)
+
+    def test_valid_barcode_scan(self):
+        barcode = self.veranstaltung.get_barcode_number()
+        response = self.client.post(self.path, {
+            "barcode": barcode,
+            "scanner_token": self.barcode_scanner.token
+        })
+        self.assertJsonSuccess(response, True)
+        self.veranstaltung.refresh_from_db()
+        self.assertEqual(self.veranstaltung.status, Veranstaltung.STATUS_GEDRUCKT)
+
+    def test_double_scan(self):
```

Comment:
This testsetup is incomplete: Because the scanner only can reach one state. There is no testing of the time constraint at all!